src/main/java/com/jcraft/jsch/Session.java

```diff
@@ -514,14 +514,17 @@ public void connect(int connectTimeout) throws JSchException{
 
       if(!auth){
         if(auth_failures >= max_auth_tries){
-          if(JSch.getLogger().isEnabled(Logger.INFO)){
-            JSch.getLogger().log(Logger.INFO,
+          if(jsch.getInstanceLogger().isEnabled(Logger.INFO)){
+            jsch.getInstanceLogger().log(Logger.INFO,
                                  "Login trials exceeds "+max_auth_tries);
           }
         }
-        if(auth_cancel)
-          throw new JSchException("Auth cancel");
-        throw new JSchException("Auth fail");
+        throw new JSchException((auth_cancel ? "Auth cancel"
+                : "Auth fail")
+                + " for methods '" + smethods + "'");
+//        if(auth_cancel)
+//          throw new JSchException("Auth cancel");
+//        throw new JSchException("Auth fail");
```

Comment:
I think we should just remove these commented out lines.

Reply:
This one is logging unrelated and is one of the changes I've done to the JSch sources. Seems that I've copied that to this file together with the logging related changes. If it's OK to keep the "for methods"-part in the exception I can remove the commented out code, otherwise I can revert it to the less informative original message

Reply:
Yes, I think keeping the "for methods" part in the exception is fine. We should just remove the old commented out code here.

Reply:
Done